### ISSUE TITLE
Incorrect modifier events

### ISSUE THREAD

**Issue by chrisduerr** [rust-windowing/winit#1357]
It seems like at least on Linux/X11, the new modifiers event is not working properly.

When pressing the key control, I get a 'ctrl' modifier in the glutin event. Then when I press the 'L' key, I get another modifier event with 'alt'. as pressed modifier.

Here's the full log, but it basically just boils down to some 'alt' event magically appearing without it ever getting pressed:

```
[2020-01-04 05:09] [INFO] glutin event: DeviceEvent { device_id: DeviceId(X(DeviceId(15))), event: Key(KeyboardInput { scancode: 29, state: Pressed, virtual_keycode: Some(LControl), modifiers: (empty) }) }
[2020-01-04 05:09] [INFO] glutin event: DeviceEvent { device_id: DeviceId(X(DeviceId(15))), event: ModifiersChanged(CTRL) }
[2020-01-04 05:09] [INFO] glutin event: WindowEvent { window_id: WindowId(X(WindowId(56623106))), event: KeyboardInput { device_id: DeviceId(X(DeviceId(3))), input: KeyboardInput { scancode: 29, state: Pressed, virtual_keycode: Some(LControl), modifiers: CTRL }, is_synthetic: false } }
[2020-01-04 05:09] [INFO] glutin event: DeviceEvent { device_id: DeviceId(X(DeviceId(15))), event: Key(KeyboardInput { scancode: 38, state: Pressed, virtual_keycode: Some(L), modifiers: CTRL }) }
[2020-01-04 05:09] [INFO] glutin event: DeviceEvent { device_id: DeviceId(X(DeviceId(3))), event: ModifiersChanged(ALT) }
[2020-01-04 05:09] [INFO] glutin event: WindowEvent { window_id: WindowId(X(WindowId(56623106))), event: KeyboardInput { device_id: DeviceId(X(DeviceId(3))), input: KeyboardInput { scancode: 38, state: Pressed, virtual_keycode: Some(L), modifiers: ALT }, is_synthetic: false } }
[2020-01-04 05:09] [INFO] glutin event: DeviceEvent { device_id: DeviceId(X(DeviceId(15))), event: Key(KeyboardInput { scancode: 38, state: Released, virtual_keycode: Some(L), modifiers: ALT }) }
[2020-01-04 05:09] [INFO] glutin event: WindowEvent { window_id: WindowId(X(WindowId(56623106))), event: KeyboardInput { device_id: DeviceId(X(DeviceId(3))), input: KeyboardInput { scancode: 38, state: Released, virtual_keycode: Some(L), modifiers: ALT }, is_synthetic: false } }
[2020-01-04 05:09] [INFO] glutin event: DeviceEvent { device_id: DeviceId(X(DeviceId(15))), event: Key(KeyboardInput { scancode: 29, state: Released, virtual_keycode: Some(LControl), modifiers: ALT }) }
[2020-01-04 05:09] [INFO] glutin event: WindowEvent { window_id: WindowId(X(WindowId(56623106))), event: KeyboardInput { device_id: DeviceId(X(DeviceId(3))), input: KeyboardInput { scancode: 29, state: Released, virtual_keycode: Some(LControl), modifiers: ALT }, is_synthetic: false } }
[2020-01-04 05:09] [INFO] glutin event: DeviceEvent { device_id: DeviceId(X(DeviceId(3))), event: ModifiersChanged((empty)) }
```

**Comment by chrisduerr** [comment]
As a potentially important sidenote, this was not introduced in the initial Linux/X11 modifiers PR though, since the current Alacritty master does not emit any alt modifier events.

**Comment by chrisduerr** [comment]
Bisected to be caused by 20e81695ca29dc742e610e0c12edc96655f4024d.

**Comment by chrisduerr** [comment]
I've identified some faulty behavior in the following code, seems like the modifiers got jumbled up.

https://github.com/rust-windowing/winit/blob/master/src/platform_impl/linux/x11/util/input.rs#L21-L24

**Comment by chrisduerr** [comment]
Fixed in #1358.
